### DISCUSSION TITLE
Comparison operator chaining

### DISCUSSION THREAD

**Discussion by The-Futurist** [dotnet/csharplang#6899] [Q&A]
I was recently looking at this capability for a new language after reading about Raku (which does have a lot of operarotors). In Raku we can write:

```cs
if (a < b < c > d)
   // do stuff;
```

Which is logically equivalent to:

```cs
if (a < b && b < c && c > d)
   return;
```

So I wondered if C# could benefit from adopting this. At first glance this seems like it might well be a non-breaking change because the scenarios under which it will compile today seem to be few if any (unless the operators have been overloaded perhaps...).

Given also that C# does accept this syntax (though this is nothing to do with comparison of course)

```cs
a = b = c = d;
```

then this does not seem that unreasonable.

This has some overlap with: https://github.com/dotnet/csharplang/discussions/217

Thoughts?

**Comment by HaloFour** [comment]
See: #4108

**Comment by The-Futurist** [comment]
Yes, as this points out:

> Potential confusion over a piece of code potentially having two different meanings. However, this is so unlikely as no real codebases should ever have been using a < b < c. It just isn't a reasonable code pattern today, so no one uses it or expects it to work the way it does today. Practically all users looking at this would expect it to have the semantics this proposal is suggesting.

Existing legitimate use of this might be either nil or very very rare.

**Reply by HaloFour** [reply]
> Existing legitimate use of this might be either nil or very very rare.

Agreed.  The championed proposal would only allow for the operator chaining in those cases where it would currently be a compiler error, which is likely very close to 100%.
